### ISSUE TITLE
Checkbox doesn't accept 'Enter'  as an input only spacebar

### ISSUE THREAD

**Issue by makopch-ms** [OfficeDev/office-ui-fabric-react#7918]
### Environment Information 
 
- __Package version(s)__: 6.122.0
- __Browser and OS versions__: all

### Please provide a reproduction of the bug in a codepen:
 
https://fabricweb.z5.web.core.windows.net/oufr/6.122.0/#/examples/checkbox
 
#### Actual behavior:
 
Pressing 'Enter' when the checkbox as keyboard focus doesn't do anything.
 
#### Expected behavior:
 
Pressing 'Enter' toggles the state of the checkbox

#### Priorities and help requested:
 
Are you willing to submit a PR to fix? No
 
Requested priority: Normal
 
Products/sites affected: (if applicable)

**Comment by micahgodbolt** [comment]
Ah, seems that regression came in when we changed checkbox back to using an input (rather than a button).

It works here https://fabricweb.z5.web.core.windows.net/oufr/6.81.0/#/examples/checkbox


Here's the PR #7521

Need to make sure and add a test to ensure that 'enter' still triggers this.

**Comment by micahgodbolt** [comment]
Ah, taking 1 more look at the PR it appears that this change was noticed

![image](https://user-images.githubusercontent.com/1434956/52384220-5d625400-2a31-11e9-9050-2d5bc48997c9.png)

**Comment by micahgodbolt** [comment]
As he said, this is the 'right', as in to spec, behavior

https://developer.mozilla.org/en-US/docs/Web/HTML/Element/input/checkbox

**Comment by makopch-ms** [comment]
@jspurlin did have a discussion @dzearing .  The email conversation sort of ended that we needed to continue that discussion and figure out what the best path is going forward and where we are to align.  So that was the purpose of opening this item to track it.  

Our team did do a workaround checking for the Enter during onKeyUp, but it is hacky and not something we want to keep.

**Comment by kenotron** [comment]
> TL;DR - My recommendation is to no fix this. This solution will be hacky either inside Fabric for all consumers or we can leave this non-standard behavior only inside the consumer code. 

## Workaround Recommendation
Given that we're still needing to ship products, we still need some degree of hackiness to get stuff done. I recommend that consumers who need this non-standard behavior would create a wrapper `React.Component` on top of `Checkbox` here to add that enter functionality. Hopefully this will be good enough solution, as this is leveraging the power of React itself to abstract something for consumers via composition. 

## Previous Art
@dzearing had a comment from his email threads that's relevant:

1. All native checkboxes leave the enter key non actioned to allow the submit button to be executed. Think what you expect to happen when you press enter on a Dialog or a Panel w an OK button.

2. Allowing them to use the enter key in some situations but not in others sets an ambiguous prescident for developers and creates unpredictability for users.

3. Adding unnecessary special case code bulks up the download sizes, something we should aggressively fight against.

4. Adding confusing or ambiguous special case flags creates api code smells and asymmeties.
